### PR TITLE
WT-11759 Evaluate prefetch error handling

### PR DESCRIPTION
**Pull request by clarissecheah**
Original PR here: https://github.com/wiredtiger/wiredtiger/pull/9949

This PR contains the contents of the original one , with the spelling of "optimization" changed to suit American spelling.

### CODE REVIEW

**Review by korteland — APPROVED** [review]
LGTM

**Review by mm-ng — APPROVED** [review]
LGTM

File: src/btree/bt_prefetch.c [hunk 1]
```diff
@@ -59,8 +52,9 @@ __wt_btree_prefetch(WT_SESSION_IMPL *session, WT_REF *ref)
             ret = __wt_conn_prefetch_queue_push(session, next_ref);
             if (ret == 0)
                 ++block_preload;
-            else if (ret != EBUSY)
-                WT_RET(ret);
+            else if (ret != EBUSY) {
```
Comment (luke-pearson) [inline src/btree/bt_prefetch.c:55]: No brackets around single line ifs.

File: src/include/error.h [hunk 2]
```diff
@@ -273,3 +273,17 @@
...
+
+/*
+ * WT_PREFETCH_ASSERT --
+ *  Assert an expression for prefetch if in diagnostic mode, or update the relevant statistic if
+ *  not. As pre-fetch is an optional optimization, we want to avoid crashing the application for
+ *  an error, but instead swallow errors where possible.
+ */
+#define WT_PREFETCH_ASSERT(session, exp, stat) \
+    do {                                       \
+        if (!(exp)) {                          \
```
Comment (luke-pearson) [inline src/include/error.h:285]: Also here.

File: src/btree/bt_prefetch.c (hunk 1 above)
Comment (clarissecheah) [reply src/btree/bt_prefetch.c]: Addressed here: [0667671](https://github.com/wiredtiger/wiredtiger/pull/9991/commits/06676715ee888f7c113959ea80694ffe58516b41)

**Review by luke-pearson — APPROVED** [review]
LGTM, big improvement thanks. 👍

**Pull request merged**